### DISCUSSION TITLE
Error when switching buffer with `consult-ripgrep`

### DISCUSSION THREAD

**Discussion by artemkovalyov** [minad/consult#909] [Q&A]
Hey @minad, after updating the package I started getting an error when selecting a candidate from `consult-ripgrep` results.

```lisp
Error in post-command-hook (consult--preview-post-command-hook): (void-function without-restriction)
consult--marker-from-line-column: Symbol’s function definition is void: without-restriction
```

I tried looking at what's new in the docs related to the preview functionality but couldn't make it go away. I also don't use the preview. Can you help me solve this or suggest in which direction to look?

Let me know if you need more input.

**Comment by minad** [comment]
Somehow your update didn't work properly, in particular missing the Compat package. See https://github.com/minad/consult#bug-reports

**Reply by artemkovalyov** [reply]
Alright, thanks, let me take a look and hopefully quickly get it back on track.

**Reply by artemkovalyov** [reply]
@minad, I did a full rebuild ensuring Compat is there:
![image](https://github.com/minad/consult/assets/10460752/0e50a273-9f8d-43eb-8f6a-15b761d9c4ca)

I used this repo to install it explicitly: https://github.com/emacs-compat/compat

Sadly still see the same error. Other consult commands that switch buffers work flawlessly.

**Reply by artemkovalyov** [reply]
The issue was in the missing lexical binding declaration for my old config which didn't cause troubles with earlier versions as it appears.

**Comment by artemkovalyov** [comment]
@minad, it's solved. Somehow consult config ended up in the only file without lexical binding yet enabled.
Thanks for the great tool of choice and all the support.
